api_tests/preprints/views/test_preprint_institutions_relationship.py

```diff
@@ -0,0 +1,304 @@
+        assert res.status_code == 403
+
+        update_institutions_payload = {'data': [{'type': 'institutions', 'id': institution_B._id}]}
+        res = app.put_json_api(url, update_institutions_payload, auth=write_user_with_institutional_affiliation.auth)
+        assert res.status_code == 200
+
+    def test_update_affiliated_institutions_add_admin_without_affiliation(self, app, admin_without_institutional_affiliation, url, institution_A):
+        """
+        Test that admins without affiliation cannot add institutions.
+        """
+        update_institutions_payload = {'data': [{'type': 'institutions', 'id': institution_A._id}]}
+        res = app.put_json_api(url, update_institutions_payload, auth=admin_without_institutional_affiliation.auth, expect_errors=True)
+        assert res.status_code == 403
+        assert res.json['errors'][0]['detail'] == f'User needs to be affiliated with {institution_A.name}'
+
+    def test_update_affiliated_institutions_add_admin_with_affiliation(self, app, admin_with_institutional_affiliation, preprint, url, institution_A):
+        """
+        Test that admins with affiliation can add institutions.
+        """
+        update_institutions_payload = {'data': [{'type': 'institutions', 'id': institution_A._id}]}
+        res = app.put_json_api(url, update_institutions_payload, auth=admin_with_institutional_affiliation.auth)
+        assert res.status_code == 200
+        preprint.reload()
+        assert institution_A in preprint.affiliated_institutions.all()
+
+        log = preprint.logs.latest()
+        assert log.action == 'affiliated_institution_added'
+        assert log.params['institution'] == {'id': institution_A._id, 'name': institution_A.name}
+
+    def test_update_affiliated_institutions_remove_unauthorized_user(self, app, user, preprint, url, institution_A):
+        """
+        Test that unauthorized users cannot remove institutions.
+        """
+        preprint.affiliated_institutions.add(institution_A)
+        preprint.save()
+        update_institutions_payload = {'data': []}
+        res = app.put_json_api(url, update_institutions_payload, auth=user.auth, expect_errors=True)
+        assert res.status_code == 403
+
+    def test_update_affiliated_institutions_remove_read_user(self, app, read_user_with_institutional_affiliation, preprint, url, institution_A):
+        """
+        Test that read users cannot remove institutions.
+        """
+        preprint.affiliated_institutions.add(institution_A)
+        preprint.save()
+        update_institutions_payload = {'data': []}
+        res = app.put_json_api(url, update_institutions_payload, auth=read_user_with_institutional_affiliation.auth, expect_errors=True)
+        assert res.status_code == 403
+
+    def test_update_affiliated_institutions_remove_write_user(self, app, write_user_with_institutional_affiliation, preprint, url, institution_A):
```

Comment:
Should this assert to ensure that the institution wasn't actually removed, or is that tested elsewhere?